android-plugin/src/main/java/io/ecocode/java/checks/social/GoogleTrackerRule.java

```diff
@@ -0,0 +1,153 @@
+ * information, but it is a first step towards Google Ads and hence this
+ * practice should be discouraged at early stage.
+ */
+
+@Rule(key = "SPRI002", name = "ecocodeGoogleTracker")
+public class GoogleTrackerRule extends BaseTreeVisitor implements JavaFileScanner {
+
+    private static final String ERROR_MESSAGE_GGL_TRACKER = "Using com.google.android.gms.analytics.* is a potential threat for privacy.";
+    private static final String ERROR_MESSAGE_FIREBASE_TRACKER = "Using com.google.firebase.analytics.* is a potential threat for privacy.";
+
+    @Override
+    public void scanFile(JavaFileScannerContext context) {
+        CompilationUnitTree cut = context.getTree();
+
+        GoogleTrackerImports googleTrackerImport = new GoogleTrackerImports();
+
+        for (ImportClauseTree importClauseTree : cut.imports()) {
+            ImportTree importTree = null;
+
+            if (importClauseTree.is(Tree.Kind.IMPORT)) {
+                importTree = (ImportTree) importClauseTree;
+            }
+
+            if (importTree == null) {
+                // discard empty statements, which can be part of imports
+                continue;
+            }
+
+            googleTrackerImport.collectTrackerImport(importTree);
+        }
+
+        handleResult(context, googleTrackerImport);
+        scan(cut);
+    }
+
+    private void handleResult(JavaFileScannerContext context, GoogleTrackerImports googleTrackerImport) {
+
+        //TODO FIX if firebase import is after google import in the same file
+        if (googleTrackerImport.hasTrackerImports()) {
+            googleTrackerImport.allAnalyticsImport = new ArrayList<>();
+            googleTrackerImport.allAnalyticsImport.addAll(googleTrackerImport.getGoogleTrackerImports());
+
+            googleTrackerImport.allAnalyticsImport.addAll(googleTrackerImport.getFirebaseTrackerImports());
+
+            List<AnalyticsIssue> issues = new ArrayList<>();
+            if (googleTrackerImport.hasGoogleTrackerImports()) {
+                for (ImportTree importTree : googleTrackerImport.getGoogleTrackerImports()) {
+                    issues.add(new AnalyticsIssue(importTree,ERROR_MESSAGE_GGL_TRACKER));
+                }
+            }
+
+            if (googleTrackerImport.hasFirebaseTrackerImports()) {
+                for (ImportTree importTree : googleTrackerImport.getFirebaseTrackerImports()) {
+                    issues.add(new AnalyticsIssue(importTree,ERROR_MESSAGE_FIREBASE_TRACKER));
+                }
+            }
+            for (AnalyticsIssue issue : issues) {
+                context.reportIssue(this,issue.tree,issue.desc);
+            }
+        }
+    }
+
+    public static class AnalyticsIssue {
+        private final ImportTree tree;
+        private final String desc;
+
+        AnalyticsIssue(ImportTree tree, String desc) {
+            this.tree = tree;
+            this.desc = desc;
+        }
+
+        public ImportTree getTree() {
```

Comment:
This getter is not used. Access is done directly on variable. Why not but in this case delete the getter.